src/Interpreters/TemporaryDataOnDisk.cpp

```diff
@@ -111,7 +111,7 @@ FileSegmentsHolderPtr TemporaryDataOnDisk::createCacheFile(size_t max_file_size)
         CreateFileSegmentSettings(FileSegmentKind::Temporary, /* unbounded */ true), FileCache::getCommonUser());
 
     chassert(holder->size() == 1);
-    holder->back().getKeyMetadata()->createBaseDirectory();
+    holder->back().getKeyMetadata()->createBaseDirectory(/* throw_if_failed */true);
```

Comment:
Without the flag won't we fail on first write attempt?

Reply:
Yes we will, but the exception message will be "No such file or directory: pp/aa/0", which is confusing for people, because they will not be able to easily understand that the reason for this is "no space left on device".